api/main.py

```diff
@@ -368,3 +371,120 @@ class parallelItem(BaseModel):
     except KeyError as e:
         print("KeyError: ", e)
         raise HTTPException(status_code=400)
+
+
+@app.get("/visual/{searchterm}")
+async def get_graph_for_file(
+    searchterm: str,
+    language: str,
+    selected: List[str] = Query([]),
+):
+    try:
+        db = get_db()
+        total_parallel_count = []
+        searchtype = 'category'
+        if re.search("^[A-Z][a-z]+$", searchterm):
+            searchtype = "collection"
+
+        # get a sorted list of categories to get the results in the right order
+        query_full_category_list = db.AQLQuery(
+            query=query_sorted_category_list,
+            bindVars={
+                "language": language,
+                "selected": selected,
+            },
+        )
+        category_list = {}
+        for category in query_full_category_list.result:
+            category_list.update(category)
+
+        # check if the search is for a catagory (i.e. T06) or for a collection (i.e. Tengyur)
+        if searchtype == "category":
+            filecount = get_query_files_per_category(language+"_"+searchterm)
+
+            for filename in filecount:
```

Comment:
nitpicking: I'm not sure about the `filecount` name. It sounds like a number, but it contains filenames.